arcgis-ios-sdk-samples/Geometry/Geodesic operations/GeodesicOperationsViewController.swift

```diff
@@ -0,0 +1,105 @@
+// Copyright 2018 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class GeodesicOperationsViewController: UIViewController, AGSGeoViewTouchDelegate {
+
+    @IBOutlet private var mapView:AGSMapView!
+
+    private var graphicsOverlay = AGSGraphicsOverlay()
+    private var originGraphic: AGSGraphic!
+    private var destinationGraphic: AGSGraphic!
+    private var pathGraphic: AGSGraphic!
+    let JFKAirportLocation = AGSPoint(x: -73.7781, y: 40.6413, spatialReference: AGSSpatialReference.wgs84())
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Add the source code button item to the right of navigation bar.
+        (self.navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = ["GeodesicOperationsViewController"]
+        
+        // Initialize map with imagery basemap.
+        let map = AGSMap(basemap: AGSBasemap.imagery())
+        
+        // Assign map to map view.
+        mapView.map = map
+        
+        // Add graphics overlay to the map view.
+        mapView.graphicsOverlays.add(self.graphicsOverlay)
```

Comment:
Feel free to drop the explicit reference to `self`.

Reply:
Removed `self`